yt_dlp/extractor/roya.py

```diff
@@ -0,0 +1,61 @@
+from .common import InfoExtractor
+from ..utils import ExtractorError
+
+
+class RoyaIE(InfoExtractor):
+    _VALID_URL = r'https?://roya\.tv/live-stream/(?P<id>\d+)'
+    _TESTS = [{
+        'url': 'https://roya.tv/live-stream/1',
+        'info_dict': {
+            'id': '1',
+            'title': r're:Roya TV \d{4}-\d{2}-\d{2} \d{2}:\d{2}',
+            'ext': 'mp4',
+            'is_live': True,
+            'live_status': 'is_live',
+        },
+    }, {
+        'url': 'https://roya.tv/live-stream/21',
+        'info_dict': {
+            'id': '21',
+            'title': r're:Roya News \d{4}-\d{2}-\d{2} \d{2}:\d{2}',
+            'ext': 'mp4',
+            'is_live': True,
+            'live_status': 'is_live',
+        },
+    }, {
+        'url': 'https://roya.tv/live-stream/10000',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        media_id = self._match_id(url)
+        stream_json = self._download_json(f'https://ticket.roya-tv.com/api/v5/fastchannel/{media_id}', media_id)
+
+        secured_url = None
+        if stream_json:
+            secured_url = stream_json.get('data', {}).get('secured_url')
+
+        if not secured_url:
+            raise ExtractorError('Failed to extract stream')
+
+        title = self.get_title(media_id)
```

Comment:
We can simplify this with `traverse_obj`
```suggestion
        title = traverse_obj(
            self._download_json('https://backend.roya.tv/api/v01/channels/schedule-pagination', media_id),
            ('data', 0, 'channel', lambda _, v: str(v['id']) == media_id, 'title', {str}, any))
```